### ISSUE TITLE
Potential snapshot issue on followers

### ISSUE THREAD

**Issue by korthout** [camunda-cloud/zeebe#7745]
**Describe the bug**

`zeebe-0` (a follower) showed a rapid increase in disk usage on `medic-cw-35-01f895df7-benchmark`.
<img width="300" alt="Screen Shot 2021-09-01 at 17 41 17" src="https://user-images.githubusercontent.com/3511026/131701742-1cdcb43c-b15a-469f-a35a-22af008c3d5e.png">

Zooming in on `zeebe-0`, it's number of log segments was linearly increasing (correlated to the increase of disk usage)
<img width="300" alt="Screen Shot 2021-09-01 at 17 43 15" src="https://user-images.githubusercontent.com/3511026/131702168-a72480cb-5140-4b99-9913-0679413aabfb.png">

And it wasn't making snapshots for some time (so it couldn't compact the log)
<img width="300" alt="Screen Shot 2021-09-01 at 17 42 18" src="https://user-images.githubusercontent.com/3511026/131701863-3ca5fe34-74d3-44a3-be42-57bbf9a226fb.png">

There were no role changes during this period. See [Grafana medic-cw-35-01f895df7-benchmark](http://34.77.165.228/d/I4lo7_EZk/zeebe?orgId=1&from=1630442441145&to=1630461620456&var-DS_PROMETHEUS=Prometheus&var-namespace=medic-cw-35-01f895df7-benchmark&var-pod=medic-cw-35-01f895df7-benchmark-zeebe-0&var-partition=All) for more details.

No errors are reported about it in Google Error Reporting.

Looking at [the logs](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22medic-cw-35-01f895df7-benchmark%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22medic-cw-35-01f895df7-benchmark%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22zeebe-cluster-helm%22%0A-jsonPayload.context.loggerName%3D%22org.elasticsearch.client.RestClient%22%0Aresource.labels.pod_name%3D%22medic-cw-35-01f895df7-benchmark-zeebe-2%22%0A-jsonPayload.context.loggerName%3D%22io.camunda.zeebe.broker.exporter%22;timeRange=2021-08-31T21:00:13.368Z%2F2021-08-31T23:00:13.368Z;cursorTimestamp=2021-08-31T21:03:36.853833Z?authuser=1&organizationId=669107107215&project=zeebe-io) it seems that zeebe-2 (another follower) has been trying to close it's partitions since at least 2 hours before, until it finally [reaches an OOM](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22medic-cw-35-01f895df7-benchmark%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fcomponent%3D%22broker%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Finstance%3D%22medic-cw-35-01f895df7-benchmark%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fmanaged-by%3D%22Helm%22%0Alabels.k8s-pod%2Fapp_kubernetes_io%2Fname%3D%22zeebe-cluster-helm%22%0A-jsonPayload.context.loggerName%3D%22org.elasticsearch.client.RestClient%22%0Aresource.labels.pod_name%3D%22medic-cw-35-01f895df7-benchmark-zeebe-2%22%0A-jsonPayload.context.loggerName%3D%22io.camunda.zeebe.broker.exporter%22%0Aseverity%3DERROR;timeRange=2021-08-31T20:00:13.368Z%2F2021-08-31T23:00:13.368Z;cursorTimestamp=2021-08-31T22:30:06.280774610Z?authuser=1&organizationId=669107107215&project=zeebe-io). Weirdly enough, I'm unable to see that in Grafana.

**Comment by npepinpe** [comment]
Possibly related to https://github.com/camunda-cloud/zeebe/issues/7694

**Comment by deepthidevaki** [comment]
There was an OOM around this time
```
Exception in thread "Broker-0-zb-actors-3" java.lang.OutOfMemoryError: Direct buffer memory at java.base/java.nio.Bits.reserveMemory(Bits.java:175) at java.base/java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:118) at java.base/java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:317) at io.camunda.zeebe.util.allocation.DirectBufferAllocator.allocate(DirectBufferAllocator.java:20) at io.camunda.zeebe.util.allocation.BufferAllocators.allocateDirect(BufferAllocators.java:16) at io.camunda.zeebe.dispatcher.DispatcherBuilder.initAllocatedBuffer(DispatcherBuilder.java:147) at io.camunda.zeebe.dispatcher.DispatcherBuilder.build(DispatcherBuilder.java:93) at io.camunda.zeebe.logstreams.impl.log.LogStreamImpl.openAppender(LogStreamImpl.java:280) at io.camunda.zeebe.logstreams.impl.log.LogStreamImpl.createWriter(LogStreamImpl.java:208) at io.camunda.zeebe.logstreams.impl.log.LogStreamImpl.lambda$newLogStreamBatchWriter$1(LogStreamImpl.java:115) at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122) at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:94) at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:78) at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:191)
``` 
As a result transition to new role was not completed. 

Closing as this might be fixed by #7744
